### PR TITLE
MAT-7084: Action Center Delete Button

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE PR

Jira Ticket: [MAT-7084](https://jira.cms.gov/browse/MAT-7084)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

**Comment by ethankaplan** [comment]
Ignoring additional coverage for the changes that are coming in 7085

File: src/api/useMeasureServiceApi.tsx [hunk 1]
```diff
@@ -96,6 +96,18 @@ export class MeasureServiceApi {
     });
   }
 
+  async deleteMeasure(measure: Measure): Promise<Response> {
+    return await axios.put(
```
Comment (adongare) [inline src/api/useMeasureServiceApi.tsx]: Not sure how this is working. endpoint is DELETE but this is PUT call. this should be `axios.delete`

File: src/api/useMeasureServiceApi.tsx [hunk 2]
```diff
@@ -96,6 +96,18 @@ export class MeasureServiceApi {
...
+      `${this.baseUrl}/measures/${measure.id}/delete`,
+      measure,
```
Comment (adongare) [inline src/api/useMeasureServiceApi.tsx]: no longer need to pass `measure` as service endpoint is not accepting it. method signature can be changed to just accept the measure.id e.g. `async deleteMeasure(id: string): Promise<Response>`

File: src/components/measureLanding/measureList/MeasureList.tsx [hunk 3]
```diff
@@ -815,6 +818,35 @@ export default function MeasureList(props: {
...
+        setToastMessage(errorMessage);
+        setToastOpen(true);
+        setDeleteMeasureDialog(false);
+      } else {
+        setToastType("danger");
+        setToastMessage(e.toString());
+        setToastOpen(true);
+        setDeleteMeasureDialog(false);
+      }
+    }
```
Comment (adongare) [inline src/components/measureLanding/measureList/MeasureList.tsx]:
```suggestion
    } catch (e) {
      if (e?.response?.data) {
        const { error, status, message } = e.response.data;
        const errorMessage = `${status}: ${error} ${message}`;
        setToastMessage(errorMessage);
      } else {
        setToastMessage(e.toString());
      }
        setToastType("danger");
        setToastMessage(e.toString());
        setToastOpen(true);
        setDeleteMeasureDialog(false);
    }
```

**Review by adongare — APPROVED** [review]

**Pull request merged**